domain/model/state/modelstate.go

```diff
@@ -355,100 +352,118 @@ WHERE value = $dbContainerTypeValue.value
 		return errors.Capture(err)
 	}
 
-	insertModelConstraintStmt, err := s.Prepare(
+	insertModelConstraintStmt, err := preparer.Prepare(
 		"INSERT INTO model_constraint (*) VALUES ($dbModelConstraint.*)",
 		dbModelConstraint{},
 	)
 	if err != nil {
 		return errors.Capture(err)
 	}
 
-	insertConstraintStmt, err := s.Prepare(
+	insertConstraintStmt, err := preparer.Prepare(
 		`INSERT INTO "constraint" (*) VALUES($dbConstraintInsert.*)`,
 		constraintInsertValues,
 	)
 	if err != nil {
 		return errors.Capture(err)
 	}
 
-	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		modelUUID, err := s.getModelUUID(ctx, tx)
-		if err != nil {
-			return errors.Errorf("getting model uuid: %w", err)
-		}
+	modelUUID, err := getModelUUID(ctx, preparer, tx)
```

Comment:
We could technically move this fetching down a lot further, closer to where it is used now?